sql/core/src/main/scala/org/apache/spark/sql/KeyValueGroupedDataset.scala

```diff
@@ -230,16 +231,19 @@ class KeyValueGroupedDataset[K, V] private[sql](
    *
    * @tparam S The type of the user-defined state. Must be encodable to Spark SQL types.
    * @tparam U The type of the output objects. Must be encodable to Spark SQL types.
+   * @param func Function to be called on every group.
    *
-   * See [[Encoder]] for more details on what types are encodable to Spark SQL.
+   *             See [[Encoder]] for more details on what types are encodable to Spark SQL.
```

Comment:
yeah. will take care of those. I have been adding more tests and restructuring the code to make it more testable.